res_partner_attributes_add_SIN/models/res_partner.py

```diff
@@ -34,7 +17,7 @@ class ResPartner(models.Model):
         help="Social Insurance Number (9 digits)",
     )
 
-    @api.one
+    @api.multi
     @api.onchange("sin")
     @api.constrains("sin")
     def validate_sin(self):
```

Comment:
@JulieSFL I'm not sure that the `@api.onchange` is really useful. If we use the method only as a constrain we would be able to use a more standard name for the method like `_check_sin(self)`. Also, I'm not very found on onchange methods that raises exceptions.